core/geometry/src/curve/internalContexts/EllipticalArcApproximationContext.ts

```diff
@@ -0,0 +1,440 @@
+
+import { assert, OrderedComparator, OrderedSet } from "@itwin/core-bentley";
+import { Geometry } from "../../Geometry";
+import { Angle } from "../../geometry3d/Angle";
+import { AngleSweep } from "../../geometry3d/AngleSweep";
+import { Range1d } from "../../geometry3d/Range";
+import { Arc3d } from "../Arc3d";
+
+/** @packageDocumentation
+ * @module Curve
+ */
+
+/**
+ * Enumeration of methods used by [[EllipticalArcApproximationContext.sampleFractions]] to return locations along each
+ * quadrant of the elliptical arc.
+ * * Because ellipses have two axes of symmetry, samples are computed for one quadrant and reflected across each
+ * axis to the other quadrants. Any samples that fall outside the arc sweep are filtered out.
+ * @internal
+ */
+export enum EllipticalArcSampleMethod {
+  /** Generate n samples uniformly interpolated between the min and max parameters of a full ellipse quadrant. */
+  UniformParameter = 0,
+  /** Generate n samples uniformly interpolated between the min and max curvatures of a full ellipse quadrant. */
+  UniformCurvature = 1,
+  /**
+   * Generate n samples interpolated between the min and max curvatures of a full ellipse quadrant, using a monotone
+   * callback function from [0,1]->[0,1] to generate the interpolation weights.
+   */
+  NonUniformCurvature = 2,
+  /**
+   * Generate samples by subdividing parameter space until the interpolating linestring has less than a given max
+   * distance to the elliptical arc.
+   */
+  SubdivideForChords = 3,
+  /**
+   * Generate samples by subdividing parameter space until the interpolating arc chain has less than a given max
+   * distance to the elliptical arc.
+   */
+  SubdivideForArcs = 4,
+};
+
+/**
+ * A monotone function that maps [0,1] onto [0,1].
+ * @internal
+ */
+export type FractionMapper = (f: number) => number;
+
+/**
+ * Options for [[EllipticalArcApproximationContext.sampleFractions]]
+ * @internal
+ */
+export class EllipticalArcApproximationOptions {
+  private _sampleMethod: EllipticalArcSampleMethod;
+  private _structuredOutput: boolean;
+  private _numSamplesInQuadrant: number;
+  private _maxError: number;
+  private _remapFunction?: FractionMapper;
+  private constructor(method: EllipticalArcSampleMethod, structuredOutput: boolean, numPointsInQuadrant: number, maxError: number, remapFunction?: FractionMapper) {
+    this._sampleMethod = method;
+    this._structuredOutput = structuredOutput;
+    this._numSamplesInQuadrant = numPointsInQuadrant;
+    this._maxError = maxError;
+    this._remapFunction = remapFunction;
+  }
+  /**
+   * Construct options with optional defaults.
+   * @param method sample method, default [[EllipticalArcSampleMethod.UniformParameter]].
+   * @param structuredOutput output format, default false.
+   * @param numSamplesInQuadrant samples in each full quadrant for interpolation methods, default 4.
+   * @param maxError max distance to ellipse for subdivision methods, default 1cm.
```

Comment:
1 cm? Really? Do we have units? I thought these numbers are unitless! I mean when we say a length is 1, then it can be 1 cm or 1 inch or whatever!

Reply:
In iTwin and imodel-native, coordinates are assumed to be in meters.

Reply:
oh nice!